ui/components/app/signature-request/signature-request.js

```diff
@@ -0,0 +1,355 @@
+import SecurityProviderBannerMessage from '../security-provider-banner-message';
+import LedgerInstructionField from '../ledger-instruction-field';
+import ContractDetailsModal from '../modals/contract-details-modal';
+import { MetaMetricsContext } from '../../../contexts/metametrics';
+import { MetaMetricsEventCategory } from '../../../../shared/constants/metametrics';
+import { SECURITY_PROVIDER_MESSAGE_SEVERITY } from '../../../../shared/constants/security-provider';
+
+import {
+  TextAlign,
+  TextColor,
+  TextVariant,
+  Size,
+  ///: BEGIN:ONLY_INCLUDE_IN(build-mmi)
+  IconColor,
+  BackgroundColor,
+  Display,
+  BlockSize,
+  ///: END:ONLY_INCLUDE_IN
+} from '../../../helpers/constants/design-system';
+import {
+  BUTTON_VARIANT,
+  Button,
+  ButtonLink,
+  TagUrl,
+  Text,
+  ///: BEGIN:ONLY_INCLUDE_IN(build-mmi)
+  Icon,
+  IconName,
+  ///: END:ONLY_INCLUDE_IN
+} from '../../component-library';
+
+///: BEGIN:ONLY_INCLUDE_IN(build-mmi)
+// eslint-disable-next-line import/order
+import Box from '../../ui/box/box';
+import { ENVIRONMENT_TYPE_NOTIFICATION } from '../../../../shared/constants/app';
+import { getEnvironmentType } from '../../../../app/scripts/lib/util';
+import { mmiActionsFactory } from '../../../store/institutional/institution-background';
+///: END:ONLY_INCLUDE_IN
+
+import Message from './signature-request-message';
+import Footer from './signature-request-footer';
+
+const SignatureRequest = ({ txData }) => {
+  const trackEvent = useContext(MetaMetricsContext);
+  const dispatch = useDispatch();
+  const t = useI18nContext();
+
+  const [hasScrolledMessage, setHasScrolledMessage] = useState(false);
+  const [showContractDetails, setShowContractDetails] = useState(false);
+  const [messageRootRef, setMessageRootRef] = useState(null);
+  const [messageIsScrollable, setMessageIsScrollable] = useState(false);
+
+  const {
+    id,
+    type,
+    msgParams: { from, data, origin, version },
+  } = txData;
+
+  // not forwarded to component
+  const hardwareWalletRequiresConnection = useSelector((state) =>
+    doesAddressRequireLedgerHidConnection(state, from),
+  );
+  const { chainId, rpcPrefs } = useSelector(getProviderConfig);
+  const unapprovedMessagesCount = useSelector(getTotalUnapprovedMessagesCount);
+  const subjectMetadata = useSelector(getSubjectMetadata);
+  const isLedgerWallet = useSelector((state) => isAddressLedger(state, from));
+  const { handleCancelAll } = useRejectTransactionModal();
+
+  ///: BEGIN:ONLY_INCLUDE_IN(build-mmi)
+  // Used to show a warning if the signing account is not the selected account
+  // Largely relevant for contract wallet custodians
+  const selectedAccount = useSelector(getSelectedAccount);
+  const mmiActions = mmiActionsFactory();
+  const accountType = useSelector(getAccountType);
+  const isNotification = getEnvironmentType() === ENVIRONMENT_TYPE_NOTIFICATION;
+  const allAccounts = useSelector(accountsWithSendEtherInfoSelector);
```

Comment:
One thing to consider -> `mapToState` relied on shallow, but equality `useSelector` relies on strict equality to trigger a re-render.

This means that if we are returning an object/array (which is the case here), whenever there's an update to the state, this component will re-render.

Not sure if this will have an impact on this component performance, buut maybe we could use update `accountsWithSendEtherInfoSelector` to use reselect to create a memoized selector?
[react-redux docs](https://react-redux.js.org/api/hooks#equality-comparisons-and-updates)